Controller/Cart/Email.php

```diff
@@ -74,6 +88,11 @@ public function execute()
                 throw new LocalizedException(__('No email received.'));
             }
 
+            if (!$this->cartHelper->validateEmail($email)) {
+                $this->bugsnag->notifyError('Invalid email address', "quote_id: {$quote->getId()} email: $email");
```

Comment:
what is the purpose of this logging?

Reply:
note bolt currently allows you to enter invalid email

Reply:
You can save invalid mail to Magento quote entity as well (It is assigned to the quote itself, not quote shipping / billing address). The validation is done later, on order creation, towards email assigned to the address. This address email may or may not be the one we have previously explicitly assigned to the quote. It depends on the user interaction flow with the system. There is a validation step before the point where our plugin assigns the address email and the address email is left unchanged (empty) if invalid. Empty address email does not break the order creation. Moreover, the validation is not consistent on all Magento versions, which implies there can be some false negatives. The purpose of the log is to save to the quote whatever email the user had entered, and have a notification of Magento marked it invalid if we by any chance need it, like some Exception was raised at a later point. This could not be a bugsnag breadcrumb or additional metadata set because it is done asynchronously from the main flow.

Reply:
I agree with Deniel, magento validate the email address.